components/molecules/InsightHeader/insight-header.tsx

```diff
@@ -22,9 +23,15 @@ const InsightHeader = ({ insight, repositories }: InsightHeaderProps): JSX.Eleme
         <ContextThumbnail size={120} ContextThumbnailURL={""}></ContextThumbnail>
       </div>
       <div className="header-info flex flex-col grow justify-center p-2">
-        <Title level={1} className="!text-3xl font-semibold tracking-tight text-slate-900">
-          {insight && insight.name || "Insights"}
-        </Title>
+        <div className="flex items-baseline">
+          <Title level={1} className="!text-3xl font-semibold tracking-tight text-slate-900 mr-[1rem]">
+            {insight && insight.name || "Insights"}
+          </Title>
+          {insight &&   
+          <Text className="flex text-xl self-end !text-light-slate-11 border border-light-slate-6 rounded-[26px] bg-light-slate-3 px-2.5 py-1 h-fit leading-3.5">
```

Comment:
This looks good visually, but the implementation can be enhanced a bit... instead of styling the `Text` component to act as a wrapper, wrapping the text in a div would reduce the `classNames` used here.

```suggestion
      <div className="flex  px-2.5 py-1 border border-light-slate-6 bg-light-slate-3 rounded-[26px]">
            // Text component can go in here
      </div>
```